### DISCUSSION TITLE
I Have 60+ blog posts but tina cms fetching only 50 posts.

### DISCUSSION THREAD

**Discussion by manpreets99** [tinacms/tinacms#4415] [General]
import { Layout } from "../../components/Layout";
import Link from "next/link";
import { useTina } from "tinacms/dist/react";
import { client } from "../../tina/__generated__/client";

export default function PostList(props) {
  // data passes though in production mode and data is updated to the sidebar data in edit-mode
  const { data } = useTina({
    query: props.query,
    variables: props.variables,
    data: props.data,
  });
  const postsList = data.postConnection.edges;
  console.log(postsList);
  return (
    <Layout>
      <h1>Posts</h1>
      <div>
        {postsList.map((post) => (
          <div key={post.node.id}>
            <Link href={`/posts/${post.node._sys.filename}`}>
              <a> manpreet {post.node._sys.filename}</a>
            </Link>
          </div>
        ))}
      </div>
    </Layout>
  );
}

export const getStaticProps = async () => {
  const { data, query, variables } = await client.queries.postConnection();
  return {
    props: {
      data,
      query,
      variables,
      //myOtherProp: 'some-other-data',
    },
  };
};

**Comment by manpreets99** [comment]
please tell me solution

**Reply by jamespohalloran** [reply]
https://tina.io/docs/graphql/queries/advanced/pagination/
